pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -549,8 +556,7 @@ func getLocalIP() ([]v1.NodeAddress, error) {
 					if ipnet.IP.To4() != nil {
 						// Filter external IP by MAC address OUIs from vCenter and from ESX
 						var addressType v1.NodeAddressType
-						if strings.HasPrefix(i.HardwareAddr.String(), MacOuiVC) ||
-							strings.HasPrefix(i.HardwareAddr.String(), MacOuiEsx) {
+						if vmwareOUI[i.HardwareAddr.String()] {
```

Comment:
This is no longer validating based on prefix. It will fail if validated against a full mac address.

Reply:
good catch @joshrosso